### PR TITLE
space is less cold

### PR DESCRIPTION
**Pull request by Sparkezel**
## About The Pull Request
Since LINDA was fixed, station has been mostly reaching the state of a frozen wasteland of airlocks, red light and alarms, this pr aims to fix that and https://github.com/tgstation/tgstation/issues/56449 and https://github.com/tgstation/tgstation/issues/56066

## Why It's Good For The Game
I already had enough of one hole turning the whole station into an icicle. Most likely there is a better way to do that but I am grasping straws here.

## Changelog
:cl: SparkezelPL
tweak: Vacuum of space is less cold
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm [hunk 1]
```diff
@@ -55,8 +55,10 @@
 	var/datum/gas_mixture/air_contents = airs[1]
 	var/datum/gas_mixture/partner_air_contents = partner.airs[1]
 
-	var/air_heat_capacity = air_contents.heat_capacity()
-	var/other_air_heat_capacity = partner_air_contents.heat_capacity()
+	var/air_heat_capacity = air_contents.heat_capacity()	if(istype(partner,/turf/open/space))
```
Comment (Fikou) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:58]: lol

File: code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm [hunk 2]
```diff
@@ -55,8 +55,10 @@
 	var/datum/gas_mixture/air_contents = airs[1]
 	var/datum/gas_mixture/partner_air_contents = partner.airs[1]
 
-	var/air_heat_capacity = air_contents.heat_capacity()
```
Comment (Fikou) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:58]: why remove this

Comment (Sparkezel) [reply code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:58]: true actually, I could've done that much better, moment

File: code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm [hunk 3]
```diff
@@ -56,7 +56,8 @@
 	var/datum/gas_mixture/partner_air_contents = partner.airs[1]
 
 	var/air_heat_capacity = air_contents.heat_capacity()
-	var/other_air_heat_capacity = partner_air_contents.heat_capacity()
+	if(istype(partner,/turf/open/space))
+		other_air_heat_capacity = 7000 ///neccesarry to keep engi and atmos in one piece but it can be changed too ;)
```
Comment (Fikou) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:60]: this shouldnt be an autodoc comment, and you never define this var

File: code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm [hunk 4]
```diff
@@ -57,6 +57,8 @@
 
 	var/air_heat_capacity = air_contents.heat_capacity()
 	var/other_air_heat_capacity = partner_air_contents.heat_capacity()
+	if(istype(partner,/turf/open/space))
+		var/other_air_heat_capacity = 7000 //neccesarry to keep engi and atmos in one piece but it can be changed too ;)
```
Comment (Jared-Fogle) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:61]: Make this magic number a define.

Comment (Fikou) [inline code/modules/atmospherics/machinery/components/unary_devices/heat_exchanger.dm:61]: you defined this twice

**Comment by Tlaltecuhtli** [comment]
👋  frostpunk 👋 bye bye 👋

File: code/__DEFINES/atmospherics.dm [hunk 5]
```diff
@@ -90,8 +90,8 @@
 #define OPEN_HEAT_TRANSFER_COEFFICIENT		0.4
 /// a hack for now
 #define WINDOW_HEAT_TRANSFER_COEFFICIENT	0.1
-/// a hack to help make vacuums "cold", sacrificing realism for gameplay
-#define HEAT_CAPACITY_VACUUM				2000
+/// space is much more realistic now, still cold as fuck but that can be "radiation"
```
Comment (Jared-Fogle) [inline code/__DEFINES/atmospherics.dm:93]: Comments don't need to explain what code used to be. Explain what it is now, especially in a doc comment.

**Comment by LemonInTheDark** [comment]
I'll do some more testing and go into more detail on this tomorrow, but I don't think this is a good idea, since it doesn't let breaches self seal, and will lead to slow vents of the station. I'm of the opinion that the issue causing station wide freezing is partially firelocks being bad at sealing off areas, and partially the simulation allowing minor diffs to keep large groups alive, which sorta breaks some things we rely on.

**Comment by Pugie** [comment]
firelocks dont fix anything even with their existence they serve as a short term annoyance to people and they open them anyways.
The issue causing station wide freezing is probably atmos being too fast

**Comment by Sparkezel** [comment]
>firelocks dont fix anything even with their existence they serve as a short term annoyance to people and they open them anyways.
The issue causing station wide freezing is probably atmos being too fast

Pretty much yes, if there was a breach anywhere, atmos would make rooms very cold because space had a lot of heat capacity per tile, because someone wanted to make space cold like in action movies and cartoons. Pressure should be the real killer imo

**Comment by Jared-Fogle** [comment]
@LemonInTheDark Any update?

**Comment by LemonInTheDark** [comment]
Yes.

I'm happy with how this works with large holes, from bombs and such. It triggers firelocks at a reasonable rate, and while the area around the breaches gets cold, the amount of mols drops quick enough that they aren't that resistant to change.

I do not like how it works with smaller breaches, but that's down to a conflict of goals between me and spark. 
Part of it is the amount of active turfs it causes, and how many parts of the station it can effect. If there's a hole in say the upper meta hallway around about 800 turfs eventually join up in one large excited group, not including churn from doors opening and closing, as there's nothing we can do about that.

I don't like being stuck simulating that large of an area with that much stuff theoretically going on. Especially if the source of those changes can take a very long time to dissipate or seal.

That's part of why I want to keep the firelock behavior, it seals off breaches like this and prevents station wide simulation, it has the added benefit of making atmos seem faster then it is, something something good for le immersion?

Like I said I disagree with part of sparks goals, he wants smaller breaches to not cool down the station to any large degree, while I want them to to allow for less processing and more reaction from the game when a hole in the station is made.  

As I've said before I think part of the issue here is firelocks poorly telegraphing where the problem areas are, and failing to seal them off from the rest of the station. Another aspect of it might also be the speed at which excited groups equalize, I initially lowered it significantly, but in light of some other structural changes it might be worth raising by a bit.

All that said, I do think space is a bit too cold. Mostly as it relates to refilling rooms. As it stands now, space drains slow enough and chills fast enough that it's very easy to end up with rooms that have too much cold gas to reheat, and while venting is an option, it's an obscure one.  

> 
> 
> firelocks dont fix anything even with their existence they serve as a short term annoyance to people and they open them anyways.
> The issue causing station wide freezing is probably atmos being too fast

Partially yes, if excited groups didn't equalize much at all over large groups we wouldn't be seeing this behavior, since that's about the only time space gets the chance to cool things down. I'm not personally willing to compromise on the atmos working quickly aspect of things.

You're right about firelocks too, as I said above they don't solve the problem because the solutions to them from a player's prospective, either toggle all the locks long enough to walk past, crow bar them up, or disable the alarms all cause more gas to spread. This is something that will need addressing no matter what happens here

**Comment by Sparkezel** [comment]
Can you test-merge it on terry and bagel please, @LemonInTheDark ? I wonder how it will work out on live

**Review by Rohesie — CHANGES_REQUESTED** [review]
Remove the changes to `.vs/slnx.sqlite` please.

**Comment by Sparkezel** [comment]
Oh god how did that slip trough, is there something I can do so it will be ignored in the future too or not really?

**Comment by LemonInTheDark** [comment]
Do you know why it's being changed?

**Comment by Sparkezel** [comment]
rather than changed for some reason when i use VS code that file is always created and like always its auto ticked so it just gets committed if I dont look

**Comment by Rohesie** [comment]
This is up to @LemonInTheDark 
Choose between testing it, approving or denying it, really.

**Comment by Sparkezel** [comment]
We are testing it on terry currently, people stopped whining about coldmos and rounds last longer, I think its good

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I'm not willing to accept something that nukes space to this degree. Firelocks closing when a breach happens isn't bad behavior, and all you're doing by lowering the heat capacity of space is increasing the time until firelocks close. 

Part of this is the result of how excited groups work, but in the end it means actual station wide freezes from unattended holes, rather then just the area inside a firelock fully chilling, with the downside of more pain on the system itself.

I'm alright with lowering spaces heat capacity just a tad, enough to make reheating a bit easier, but this is way too much. I understand if you're not willing to do that, since it would be like stamping your name on the current state of things. Let me know, and based on that I'll close/keep the pr open.

I got the chance to watch some of the testmerge, and saw these problems crop up. I still think the way to solve station wide cold is to make cold vs warm areas more distinct, lower the amount of alarm fatigue around fire alarms, and cut down on some of the more unpleasant sources of churn on the station, so large excited groups don't stick around forever.

**Comment by LemonInTheDark** [comment]
Keep in mind, the lower that number is the longer it will take for fire alarms to get cold enough to seal off a breach. 3500 is way too low, and I'm not willing to merge it at this value

**Comment by Sparkezel** [comment]
yeah that was for tests
gonna bump it to 5400, that should be fine I guess

**Comment by LemonInTheDark** [comment]
Need to test this more, oh and uh, if you're not going that much lower then current it might be better to just keep things as one define, and take the hit on heat exchangers, for sanity and cleanliness

**Comment by moocowswag** [comment]
Will this make the round start cooling loop of the sm less effective since its partially reliant on space? I cant imagine it changing much but it could lead to a surge in engineer failures.

**Comment by DrCrawler** [comment]
Why make spess less cold when you can just make gas not take the coldness from space. Atleast lessen the sp- Wait, how can no gas have heat capacitance? Dude just nerf the speed of gas being cooled directly from space. Its easyer and less messy (from my opinion). Heat capacitanse is how hard it is to cool/heat a gas or a mixture of gasses. Space is emty so by logic heat capacitanse in it shouldn't even exist. Space realy should cool the air by decompression but thats a different thing for a different pull. If im dumb just explain where did i say something wrong because im a chemist, not a physician and thus dont realy understand the stuff that good. Yeah and also BYOND doesn't fully simulate the physics so you might as well give me an idea how does space cool gas.

**Comment by Mothblocks** [comment]
what

**Comment by DrCrawler** [comment]
what

**Comment by LemonInTheDark** [comment]
> 
> 
> Why make spess less cold when you can just make gas not take the coldness from space. Atleast lessen the sp- Wait, how can no gas have heat capacitance? Dude just nerf the speed of gas being cooled directly from space. Its easyer and less messy (from my opinion). Heat capacitanse is how hard it is to cool/heat a gas or a mixture of gasses. Space is emty so by logic heat capacitanse in it shouldn't even exist. Space realy should cool the air by decompression but thats a different thing for a different pull. If im dumb just explain where did i say something wrong because im a chemist, not a physician and thus dont realy understand the stuff that good. Yeah and also BYOND doesn't fully simulate the physics so you might as well give me an idea how does space cool gas.

buddy space is just really cold, and we fake the heat capacity for both gameplay and code cleanliness

**Comment by optimumtact** [comment]
@LemonInTheDark you have 48 hours to either move on this or I close it

**Comment by LemonInTheDark** [comment]
I'm going to close this.

I asked for the minor lowering of heat capacity because I was hoping it would make cold spaces easier to deal with, but I've since realized that we're better off giving people better tools, especially since more things rely on space being cold, and I'm reasonably sure even more will soon. I'm not willing to do the "one heat capacity for space, one for he pipes" solution either, as that's much too hacky.

I'm going to keep going with my plan of improving atmos tech's control of their environment, and encouraging fixing through gameplay. We'll see how things work out.

**Pull request closed without merging**